openeo/local/connection.py

```diff
@@ -0,0 +1,316 @@
+import datetime
+import xarray as xr
+import numpy as np
+import rioxarray
+from glob import glob
+from pathlib import Path
+from pyproj import Transformer
+from typing import Dict, List, Tuple, Union, Callable, Optional, Any, Iterator
+
+from openeo.metadata import CollectionMetadata
+from openeo.internal.graph_building import PGNode
+from openeo.rest.datacube import DataCube
+from openeo.internal.jupyter import VisualDict, VisualList
+
+def _get_temporal_dimension(dims):
+    if 't' in dims:
```

Comment:
fyi you can do this more compactly like 

```python
for field in ["t", "time", "temporal", "DATA"]:
    if field in dims:
        return field
```
(explicit `None` return at the end is optional)

Reply:
You are totally right, thanks for the hint.

Reply:
you can make it even generic, instead of a separate function for each dimension:

```python
def _get_dimension(dims: dict, candidates: List[str]):
    for name in candidates:
        if name in dims:
            return name
....

x_dim = _get_dimension(data.dims, ["x", "X", "lon", "longitude"])
...
```